src/category_theory/adjunction/lifting.lean

```diff
@@ -0,0 +1,288 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.limits.shapes.equalizers
+import category_theory.adjunction
+import category_theory.monad.adjunction
+
+/-!
+# Adjoint lifting
+
+This file gives two constructions for building left adjoints: the adjoint triangle theorem and the
+adjoint lifting theorem.
+The adjoint triangle theorem says that given a functor `U : B ⥤ C` with a left adjoint `F` such
+that `ε_X : FUX ⟶ X` is a coequalizer for the pair `(FUε_X, ε_FUX)` for every `X` (where `ε` is the
+counit of the adjunction `F ⊣ U`). Then for any category `A` with coequalizers of reflexive pairs,
+a functor `R : A ⥤ B` has a left adjoint if the composite `R ⋙ U` does.
+Note that the condition on `U` regarding `ε_X` is automatically satisfied in the case when `U` is
+a monadic functor, giving the corollary: `monadic_adjoint_triangle_lift`, i.e. if `U` is monadic,
+`A` has reflexive coequalizers then `R : A ⥤ B` has a left adjoint provided `R ⋙ U` does.
+
+The adjoint lifting theorem says that given a commutative square of functors (up to isomorphism):
+
+    Q
+  A → B
+U ↓   ↓ V
+  C → D
+    R
```

Comment:
I think this will look bad on the web docs unless you indent it into a code block
```suggestion
        Q
      A → B
    U ↓   ↓ V
      C → D
        R
```